src/components/Timer/Timer.tsx

```diff
@@ -27,22 +29,27 @@ const Timer: React.FC<Props> = (timerProps) => {
 
 	useEffect(() => {
 		if (seconds <= 0) {
-			onTimeExpiration();
+			if (!expired) {
+				setExpired(true);
+				onTimeExpiration();
+			}
 		} else {
 			paused
 				? onPause(seconds)
 				: (timer.current = window.setInterval(() => setSeconds((v) => v - 1), 1000));
 		}
 		return () => clearInterval(timer);
-	}, [seconds, onTimeExpiration, clearInterval, paused, onPause]);
+	}, [seconds, clearInterval, paused, onPause, expired, onTimeExpiration]);
 
 	const timerStyle: CSSProperties = {
 		'--deg': `${360 - (360 / duration) * seconds} `,
-		'--col': `hsla(${(100 / duration) * seconds}, 100%, ${50 - duration / seconds}%, 1)`,
+		'--col': `hsla(${(100 / duration) * seconds}, 100%, ${
+			50 - duration / Math.max(1, seconds)
+		}%, 1)`,
 	};
 
 	return (
-		<div className="timer" style={timerStyle}>
+		<div className={classNames('timer', { expired: expired })} style={timerStyle}>
```

Comment:
nip: если ключ и название переменной совпадает, то можно писать чисто название
```suggestion
		<div className={classNames('timer', { expired })} style={timerStyle}>
```

Reply:
удобно, поправлю